lib/src/main/java/com/palantir/computemodules/auth/ThirdPartyAuth.java

```
@@ -0,0 +1,120 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.computemodules.auth;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.palantir.logsafe.Unsafe;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+public final class ThirdPartyAuth {
+
+    private static final SafeLogger log = SafeLoggerFactory.get(ThirdPartyAuth.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private ThirdPartyAuth() {}
+
+    public static ThirdPartyCredentials retrieveThirdPartyIdAndCreds() {
+        return new ThirdPartyCredentials(System.getenv("CLIENT_ID"), System.getenv("CLIENT_SECRET"));
+    }
+
+    @Unsafe
+    public static Optional<String> fetchOAuthToken(String hostname, List<String> scope) {
+        ThirdPartyCredentials credentials = retrieveThirdPartyIdAndCreds();
+        try {
+            String formData = "grant_type=" + URLEncoder.encode("client_credentials", StandardCharsets.UTF_8)
+                    + "&client_id=" + URLEncoder.encode(credentials.clientId().orElseThrow(), StandardCharsets.UTF_8)
```

should we really throw here? I think we should throw a better explanation saying credentials are not present or return None just like we do with python https://github.com/palantir/python-compute-module/blob/5baaba6d462a9bafb38e71375c282dcabd19f45f/compute_modules/auth/third_party.py#L25